### ISSUE TITLE
OTA not showing for IKEA E2206 (INSPELNING) after upgrade to Z2M 1.41.0

### ISSUE THREAD

**Issue by lbschenkel** [Koenkk/zigbee2mqtt#24572]
### What happened?

I have a number of IKEA E2206 (INSPELNING) plugs. They are still on factory firmware, version 2.4.34 and not updated to the latest firmware. Instead of doing workarounds to update them, I was waiting for Z2M 1.41.0 to be released because I knew that it would come with fixes to the OTA URL.

However, after upgrading to 1.41.0 and clicking on "check for new updates" for those devices, I still get "no updates available".

### What did you expect to happen?

Given that [#3067](https://github.com/Koenkk/zigbee2mqtt.io/pull/3067) was fixed and incorporated into this release, I was expecting the OTA to work now and for the new firmware to be found.

I also removed and rejoined the device, just in case, but it didn't change the outcome.

### How to reproduce it (minimal and precise)

- Update to 1.41.0
- Go to OTA page, click on "check for new updates"
- Popup with "no update available" is shown

### Zigbee2MQTT version

1.41.0

### Adapter firmware version

20240710

### Adapter

ZBDongle-P

### Setup

Home Assistant add-on

### Debug log

```
info 2024-11-01 12:18:14z2m: Checking if update available for 'REDACTED'
info 2024-11-01 12:18:16z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/REDACTED', payload '{"current":0,"energy":0,"last_seen":"2024-11-01T12:18:16+01:00","linkquality":127,"power":0,"power_on_behavior":"on","state":"ON","update":{"installed_version":33816628,"latest_version":33816628,"state":"idle"},"update_available":null,"voltage":23.2}'
info 2024-11-01 12:18:16z2m: No update available for 'REDACTED'
```

**Comment by jamesonuk** [comment]
As per
https://github.com/Koenkk/zigbee2mqtt/discussions/24464
and
https://github.com/Koenkk/zigbee-OTA/commit/35afc409fee009079ddf1f67a1163130084eb1cc


https://github.com/Koenkk/zigbee-herdsman-converters/blob/f54114a3de95b9380f895007a68809ea1a958346/src/devices/ikea.ts#L606C113-L607C18

need to be changed to use zigbeeOTA (probably needs import too)
to get the OTA you are going to either update the definition or create an external converter as in https://github.com/Koenkk/zigbee2mqtt/issues/23961#issuecomment-2451784804 until someone puts in a PR to change the converter

**Comment by randommen96** [comment]
Great workaround for the time being, thanks for the clear steps!

**Comment by jamesonuk** [comment]
Created https://github.com/Koenkk/zigbee-herdsman-converters/pull/8248 to add this

**Comment by treetip** [comment]
You can just use this custom converter to update all your plugs, and then just remove it (it's built from the current master version, just uses ota instead of ikeaOta)

```
const constants_1 = require("zigbee-herdsman-converters/lib/constants");
const ikea_1 = require("zigbee-herdsman-converters/lib/ikea");
const modernExtend_1 = require("zigbee-herdsman-converters/lib/modernExtend");

const definition = 
    {
        zigbeeModel: ['INSPELNING Smart plug'],
        model: 'E2206',
        vendor: 'IKEA',
        description: 'INSPELNING smart plug',
        extend: [(0, ikea_1.addCustomClusterManuSpecificIkeaUnknown)(), (0, modernExtend_1.onOff)(), (0, modernExtend_1.identify)(), (0, modernExtend_1.ota)(), (0, modernExtend_1.electricityMeter)()],
        configure: async (device) => {
            const endpoint = device.getEndpoint(1);
            // Enable reporting of powerDivisor, needs to change dynamically with the amount of power
            // For details, see: https://github.com/Koenkk/zigbee2mqtt/issues/23961#issuecomment-2366733453
            await endpoint.configureReporting('haElectricalMeasurement', [
                { attribute: 'acPowerDivisor', minimumReportInterval: 10, maximumReportInterval: constants_1.repInterval.MAX, reportableChange: 1 },
            ]);
        },
    };

module.exports = definition;
```

**Comment by Koenkk** [comment]
Fixed now, thanks @jamesonuk !

Changes will be available in the [dev branch](https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html) in a few hours from now.

**Comment by rolfie23** [comment]
> Fixed now, thanks @jamesonuk !
> 
> Changes will be available in the [dev branch](https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html) in a few hours from now.

Updated just few minutes ago to 1.41.0-dev commit: 848f250d when came available but still no update message. In the commit I doesn't see it mentioned as fix.

**Comment by Koenkk** [comment]
Were some issues with the update, should be ready once https://github.com/Koenkk/zigbee2mqtt/pull/24609 is merged

**Comment by rolfie23** [comment]
> Got some issues with the update, should be ready once #24609 is merged

Okay, don't know where to see if it's merged so can you let me know?

**Comment by Misiu** [comment]
> * docker exec -it addon_45df7312_zigbee2mqtt bash

can this be done from Terminal & SSH addon?
when I try to exec that command I get: command not found

![image](https://github.com/user-attachments/assets/839ff44a-cdc7-4707-8afb-7bb57782c075)

**Comment by mitch66692** [comment]
> > Got some issues with the update, should be ready once #24609 is merged
> 
> Okay, don't know where to see if it's merged so can you let me know!
> 
> EDIT: Saw there was an new dev, updated and did get an error: `Failed to check if update available for 'Inspelning stopcontact' (ENOENT: no such file or directory, open '/opt/zigbee2mqtt/data/index.json')`. After putting "https://raw.githubusercontent.com/Koenkk/zigbee-OTA/refs/heads/master/index.json" in OTA index override it was working! it's now updating!

Is it really so simple to fix it? Does OTA index override file name can have any sort of problem if we copy and paste that link?

**Comment by rolfie23** [comment]
> > > Got some issues with the update, should be ready once #24609 is merged
> > 
> > 
> > Okay, don't know where to see if it's merged so can you let me know!
> > EDIT: Saw there was an new dev, updated and did get an error: `Failed to check if update available for 'Inspelning stopcontact' (ENOENT: no such file or directory, open '/opt/zigbee2mqtt/data/index.json')`. After putting "https://raw.githubusercontent.com/Koenkk/zigbee-OTA/refs/heads/master/index.json" in OTA index override it was working! it's now updating!
> 
> Is it really so simple to fix it? Does OTA index override file name can have any sort of problem if we copy and paste that link?

It was after the dev update. Stable update next month its in, after that update did get the error message. This was the solution. Its just an file for an repository wich get updated by koenkk.

Here everything is now updated, besides of the round 5 button remote of IKEA. Gets at 100% and then says it's updated with same firmware. That I'll leave for now, have some other things to do.

**Comment by adamk125** [comment]
> Got some issues with the update, should be ready once #24609 is merged

Hi Guys,

I am still having issues updating my Inspelning Smart Plugs even with the latest version of Z2M (not the dev branch). I hoped that the updates were merged into the latest version.

Do I need to do the external converter trick for this to work?

Thanks

![image](https://github.com/user-attachments/assets/eac84bf9-351c-41d3-9698-8d1f525478f9)

![image](https://github.com/user-attachments/assets/dbed8697-68f2-450a-97b8-d37b8e438aeb)

**Comment by adamk125** [comment]
> > Got some issues with the update, should be ready once #24609 is merged
> 
> Hi Guys,
> 
> I am still having issues updating my Inspelning Smart Plugs even with the latest version of Z2M (not the dev branch). I hoped that the updates were merged into the latest version.
> 
> Do I need to do the external converter trick for this to work?
> 
> Thanks
> 
> ![image](https://private-user-images.githubusercontent.com/173307540/401687462-eac84bf9-351c-41d3-9698-8d1f525478f9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzcwMDY1ODgsIm5iZiI6MTczNzAwNjI4OCwicGF0aCI6Ii8xNzMzMDc1NDAvNDAxNjg3NDYyLWVhYzg0YmY5LTM1MWMtNDFkMy05Njk4LThkMWY1MjU0NzhmOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExNlQwNTQ0NDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MWNkMTQ2NzNlMmMxNGQ0MzdlNTJlZGQyYTc2YmE2MjFkZmQ3Yjg4ODE5OTJjNzJlZDYxMDU0ZmE0ZTIyN2U1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.P6hSTJfcw0_kFPqr2zEflLsLaWPOBw2SduTJyQlz86k)
> 
> ![image](https://private-user-images.githubusercontent.com/173307540/401687753-dbed8697-68f2-450a-97b8-d37b8e438aeb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzcwMDY1ODgsIm5iZiI6MTczNzAwNjI4OCwicGF0aCI6Ii8xNzMzMDc1NDAvNDAxNjg3NzUzLWRiZWQ4Njk3LTY4ZjItNDUwYS05N2I4LWQzN2I4ZTQzOGFlYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExNlQwNTQ0NDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YmFhMGFkZjM3OTRkOTExN2UyMDM2NGJiODU1ZDgwMjkyNWFiYTkyYjgxMTNlOWJhZDQzZWFiNjYzNGJkZGQ2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.f3J2F0e6F8Zkrg_-26KqOcmOazkq7U9iL6ydSH93Kww)
> 
> ![image](https://private-user-images.githubusercontent.com/173307540/401688085-7df44bff-ebad-47cc-b779-10d4317ac210.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzcwMDY1ODgsIm5iZiI6MTczNzAwNjI4OCwicGF0aCI6Ii8xNzMzMDc1NDAvNDAxNjg4MDg1LTdkZjQ0YmZmLWViYWQtNDdjYy1iNzc5LTEwZDQzMTdhYzIxMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExNlQwNTQ0NDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YTE2N2JmZWEzMmY5NzYxOTc1MmMzZjgyOWQ3NjczZGIxNWM0ZDQ5YzFlMDY3MmI5ZDc5YjRmOTcwYTMzN2Y1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.u76gc2sD1LUrPzczQVftQTLy8_OlMnPiINODXKxpXnA)

Issue resolved finally. See comment in this thread: https://github.com/Koenkk/zigbee2mqtt/issues/23961#issuecomment-2593997740
